code/datums/keybinding/xeno.dm

```diff
@@ -1,26 +1,407 @@
 /datum/keybinding/xeno
 	category = CATEGORY_XENO
 	weight = WEIGHT_MOB
+	var/ability_signal
 
+/datum/keybinding/xeno/down(client/user)
+	if(SEND_SIGNAL(user.mob, ability_signal) & COMSIG_XENOABILITY_HAS_ABILITY)
+		return TRUE
+
+	if(!isxeno(user.mob))
+		return
+
+	to_chat(user, "<span class='notice'>You don't have this ability.</span>") // TODO Is this spammy?
```

Comment:
yes

Reply:
it also made sharing keys for different abilities not work